src/coordination/coordinator_instance.cpp

```diff
@@ -30,7 +30,8 @@ using nuraft::ptr;
 using nuraft::srv_config;
 
 CoordinatorInstance::CoordinatorInstance()
-    : raft_state_(RaftState::MakeRaftState(
+    : thread_pool_{1},
```

Comment:
I don't see any check here about opened lock when leadership is changed?

Reply:
I added this part on new PR